strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-nuget-layout-provider/src/test/java/org/carlspring/strongbox/artifact/generator/NugetPackageGenerator.java

```diff
@@ -62,6 +66,24 @@ public NugetPackageGenerator(File basedir)
         this.basedir = basedir.getAbsolutePath();
     }
 
+    @Override
+    public Path generateArtifact(URI uri,
+                                    int size)
+            throws IOException
+    {
+        try
+        {
+            NugetArtifactCoordinates coordinates = NugetArtifactCoordinates.parse(Paths.get(uri).toString());
+
+            generateNugetPackage(coordinates.getId(), coordinates.getVersion());
+        }
+        catch(JAXBException | NoSuchAlgorithmException | NugetFormatException e)
+        {
+            throw new IOException(e);
+        }
+        return Paths.get(uri);
```

Comment:
you should provide constructor with `Path` as above and return `Path` instance pointing on package that was generated.

Reply:
@sbespalov in the `NugetPackageGenerator `class the `basePath `is of `String `type whereas in `MavenArtifactGenrator `and `NPMPackageGenerator`, it is of `Path `type. 

Should i change the type of `basePath  `to `Path `in `NugetPackageGenerator  `class?

Also I didn't understand how should I return the Path instance pointing to package that was generated? 
In the `NPMPackageGenerator `class there is a method `getPackagePath` as mentioned in the above comment. But there is no such method in `NugetPackageGenerator  `class ans also the `generateNugetPackage `method is of void type.

Reply:
@carlspring could you pls help me with the above?

Reply:
> @sbespalov in the `NugetPackageGenerator `class the `basePath `is of `String `type whereas in `MavenArtifactGenrator `and `NPMPackageGenerator`, it is of `Path `type.
> 
> Should i change the type of `basePath `to `Path `in `NugetPackageGenerator `class?
 
I think it would be best if all of these generators use `Path`. This might require some minor refactoring, of course.

> Also I didn't understand how should I return the Path instance pointing to package that was generated?
> In the `NPMPackageGenerator `class there is a method `getPackagePath` as mentioned in the above comment. But there is no such method in `NugetPackageGenerator `class and also the `generateNugetPackage `method is of void type.

Well, maybe you'll have to implement these methods, if they don't already exist.